### DISCUSSION TITLE
Only small menu with Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by loewexy** [Martchus/syncthingtray#280] [General]
Hey, I just discovered syncthingtray and installed it on my Ubuntu 24.04 system. I setup the connection and this works without a problem. However when I click the Icon I only get a very small menu like the following:

![Bildschirmfoto vom 2024-08-31 11-09-00](https://github.com/user-attachments/assets/52f5a35f-d9ef-42aa-b1e1-ba200681ba23)

If I enter `syncthingtray -w` into my terminal, I will get the Windows described in the documentation screenshots but it is not shown after a click on the tray icon. Can anyone tell my why this is the case and how it can be fixed?

**Comment by Martchus** [comment]
Saying you're using "Ubuntu 24.04" is not very helpful alone. What DE are you using? Is it by any chance Wayland-based? Then I'm afraid it is not supported by the standalone version because Wayland itself makes it impossible. There are workaround, though. I mentioned them in the README.

**Reply by loewexy** [reply]
I am using the standard Ubuntu 24.04 distribution, meaning Gnome as Desktop environment and Wayland.

In the README I do not find relevant info for wayland. It says it should be supported and that there can be positioning issues. But I get a completely different menu which I did not find in the documentation.

**Reply by Martchus** [reply]
Ok - the mentioned positioning errors will be a problem. It is just that it apparently even fails before it comes to that in your environment. 

Maybe I should clarify the wording about Wayland in the README. It is not really supported but with workarounds maybe still usable. 

I guess in you environment the problem is that your system tray only ever shows "D-Bus" menu but does never deliver the signal that the icon was activated. (The small "D-Bus" menu is normally supposed to be shown only on right clicks. The custom UI of Syncthing Tray is supposed to be shown when the tray icon is activated which is normally happening via a left click.)

**Reply by Martchus** [reply]
Note that your GNOME installation doesn't seem to be a standard one. I did some testing with standard GNOME a while ago and it had no tray area at all. Users said a GNOME extension could be used to bring it back so I mentioned that in the README. In your case also some kind of extension must be installed (unless GNOME recently gained native tray support which I highly doubt considering the endless/unresolved discussions around the new API for that).

**Reply by loewexy** [reply]
Ubuntu delivers the extension for gnome showing the tray icons.

If you want to test this yourself, fire up a new VM install standard Ubuntu Desktop and test with that. My installation is from yesterday so pretty new therefore it should be reproduceable with any fresh install.

For me it is no big problem since the tray icon state works and the small menu offers the option to show the syncthing web UI in a dedicated window, that is all I was searching for. I just was confused that it got a menu which I never saw before in any of the documentation but not the menu shown in the documentation.

**Reply by Martchus** [reply]
Ok, I kind of expected that. It is not a very good extension unfortunately. On the other hand it at least shields you completely from experiencing any positioning issues. 

And no, I don't want to test this right now. At some point one has to draw a line to not lose focus. 

I suppose it is not a big deal, indeed. I made it so that you can still access all settings and open Syncthing itself from the small menu and every part of the GUI that you can access is still supposed to function as expected. I suppose I should still clarify the wording in the README, though.

Note that while the "big menu" is hidden Syncthing Tray also only does a reduced amount of queries. So you at least also don't pay for what you don't use.

**Reply by loewexy** [reply]
Okay that confirmation is nice, adding that to the README would have been sufficient for me. Since I just wanted a simple tray icon and the possibility to open the webpage in a dedicated window with matching icon.

Also thanks for the clarification that this won't cost me anything, therefore I will now use it just for this purpose.
